agent-dispatchers/kinesis/src/main/java/com/expedia/www/haystack/agent/dispatcher/KinesisDispatcher.java

```diff
@@ -143,7 +146,17 @@ AWSCredentialsProvider buildCredsProvider(final Map<String, String> conf) {
         final Object awsAccessKey = conf.remove(AWS_ACCESS_KEY);
         final Object awsSecretKey = conf.remove(AWS_SECRET_KEY);
 
-        if (Objects.nonNull(awsAccessKey) && Objects.nonNull(awsSecretKey)) {
+        if (Objects.nonNull(awsAccessKey) && Objects.nonNull(awsSecretKey) && Objects.nonNull(stsRoleArn)) {
+            return new STSAssumeRoleSessionCredentialsProvider.Builder(stsRoleArn.toString(), "haystack-agent")
```

Comment:
can you throw some light on this change?
I think we already support assuming role functionality in this function.

Reply:
The assume role functionality below assumes that the user (or instance) is already authenticated. We need a way to also pass in credentials when assuming the role so that a given user (with a set of credentials) can assume the role.

Reply:
Are you running this outside AWS env?
Because then right IAM would solve your problem. Right?

Reply:
We have two scenarios: 1) running outside of AWS, and 2) running inside AWS but without the ability in our deployment to attach credentials to the instance we run on. Both of these needing to assume a role.

You could argue that I could just use environment variables or set Java system properties _prior_ to attempting to assume the role. That would work.
However, it makes sense that if you pass into the configuration the 3 optional parameters: `AwsAccessKey`, `AwsSecretKey`, `StsRoleArn` that it would use all of them to construct your credentials provider.

Reply:
ok, that should be fine.